### DISCUSSION TITLE
can't run :Copilot setup using LazyVim

### DISCUSSION THREAD

**Discussion by lucazz** [community/community#46866] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I'm running neovim w/ LazyVim and added a new file under: 
`~/.config/nvim/lua/plugins/copilot.lua` 

With the following content, as documented by LazyVim's documentation: 

```
return {
    {
        "github/copilot.vim",
    },
}
```

Lazy.vim installs copilot as expected:
<img width="768" alt="Screenshot 2023-02-09 at 11 54 31" src="https://user-images.githubusercontent.com/6126301/217847516-8e8d0e05-b1c2-43ed-9b78-67af3fea49d7.png">

But when I run `:Copilot setup`, neovim freezes up and I'm only able to exit it by killing it's process.

**Comment by jmiller-rise8** [comment, marked as answer]
@lucazz 
I have been able to recreate issue on my machine. 

I found a workaround.

Run `:Copilot setup` then it will look like NeoVim is frozen. ~Type `:wq<enter>`~ Press enter again then your browser should open up to the Github.com Device Activation page. The code should be in your clipboard and if you paste the code in, hit Continue, Authorize the App, it should land you on the device success page. At that point a directory should have been created on your machine, `~/.config/github-copilot/hosts.json`.

Hopefully this helps!

**Reply by lucazz** [reply]
@jmiller-rise8 Indeed that worked for me as well!

I'm going to upvote this so other people can see this, but I rather have a final answer be marked as the answer if you don't mind :-)

**Reply by jmiller-rise8** [reply]
I'm hesitant to suggest this is the final answer. It certainly works but doesn't actually fix the issue at hand. Probably wait until @tpope has given his blessing.

**Reply by tpope** [reply]
@lucazz @jmiller-rise8 does it display any output before it freezes? What happens if you then press Enter?

**Reply by lucazz** [reply]
No it doesnt 😢

**Reply by jmiller-rise8** [reply]
@tpope I updated my original comment to reflect that pressing enter after running the setup command does in fact open the GitHub auth page for activation with the code in the clipboard. There is nothing displayed in NeoVim (LazyVim). Zero error messages.

Additional info:
```
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Ventura-arm64.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.8.3/share/nvim"

Run :checkhealth for more info
```

**Reply by tpope** [reply]
I see, the issue is that noice.vim is swallowing the output.

> Run `:Copilot setup` then it will look like NeoVim is frozen. Press enter again then your browser should open up to the Github.com Device Activation page. The code should be in your clipboard and if you paste the code in, hit Continue, Authorize the App, it should land you on the device success page.

Until it's properly addressed, I endorse this workaround.

**Reply by wizcas** [reply]
YOU!! ARE!! MY!! SAVIOR!!!!! ❤️❤️❤️

**Reply by dablurr** [reply]
Seems the issue is still present as of today but your workaround @tpope did the job. Thanks !

**Reply by aaronmahlke** [reply]
the noice.vim alert is displayed but almost transparent, so very easy to miss. The workaround with pressing enter worked for me.

**Comment by DavidXu-JJ** [comment]
Hi, everyone.
I find the same issue on my own machine. And I failed to solve the problem with the above mentioned solution.
I finally solve this by upgrade my neovim from old 0.8.x version to the latest version. I enable the copilot from `:LazyExtras` first and `:Copilot auth` to get authorized. Then add the "github/copilot.vim" to the lua plugin configuration file. At last, I type `:Copilot setup` and it automatically get the authorization info from the previous `:Copilot auth`. This works for me on Linux(Ubuntu 22.04). 
For ones who also can't solve this problem from the above solution, maybe you can try to upgrade your neovim. And my solution is just for reference.

**Reply by sktzofrenic** [reply]
Yes, this worked for me. Authenticate using lazy extras then disable the extra and install copilot.

**Comment by tpope** [comment]
`:Copilot setup` on LazyVim is fixed in Copilot.vim 1.19.0.

**Comment by Gabriel-Goes** [comment]
the error persists on my machine with all plugins up to date. But the work around does in fact _works_.
